### PR TITLE
[WIP] Indicator applet

### PR DESCRIPTION
**Pull request by EduardDopler**
As most Cinnamon applets use indicators instead of system trays and
indicators are automatically themed when using certain icons, Mint
Update was updated to use an Indicator.

Sadly, the underlying indicator library can only show a menu on click,
clicks on the icon are not customizable, and there is no signal for
double-click. Hence, in order to show the main window, one needs to
click on a menu item now.
Also there is no tooltip support (who came up with such a library
anyway?) so status information will be displayed inside the menu.

For automatic theming, icons need to be suffixed with -symbolic.svg.
I designed flat icons which match the panel theme automatically.

This closes linuxmint/mintupdate#101

### CODE REVIEW

**Comment by clefebvre** [comment]
Indicator support was added to Cinnamon recently. For most people this represented a regression rather than an improvement though. It made many apps switch from statusIcons to indicators and the result according to various feedback was a loss of feature in many of the tray icons. For this reason, indicator support is now disabled by default in Cinnamon.

From a development point of view this is real pity because rendering the tray statusicon icon picture in Cinnamon is extremely hacky (it's basically cut from the underlying rendering and pasted on top of the clutter objects, leading to many size and rendering issues).

In contrast, indicator icons are much saner and can be rendered without issues.

We thought about extending libindicator in Linux Mint, or forking it into a new library our own mint-specific tools (like mintupdate) could use, and to give it tooltip and extended click/hover support. This wasn't done yet though and it needs to be considered prior to porting Mint tools onto it.

When it comes to cross-distribution applications (blueberry for instance) I don't really have a solution. We can't improve indicator usage in many of the apps and it's less satisfactory than statusIcon, so it's likely Cinnamon will continue to ship with that support disabled by default. 

We're also not able to make the fallback between indicator and statusIcon specific to particular applications. The minute support is enabled, all indicator-compatible apps switch to indicators.

Anyway, long story short, we need to discuss this more and we're not ready to port individual tools away from statusIcon just yet.

**Comment by clefebvre** [comment]
I switched to appindicator3 temporarily when running in KDE (as Plasma no longer supports status icons).
